ql/instruments/makeswaption.cpp

```diff
@@ -72,13 +73,13 @@ namespace QuantLib {
         }
 
         Rate usedStrike = strike_;
+        ext::shared_ptr<OvernightIndexedSwapIndex> OIswap_index = ext::dynamic_pointer_cast<OvernightIndexedSwapIndex>(swapIndex_);
         if (strike_ == Null<Rate>()) {
             // ATM on curve(s) attached to index
             QL_REQUIRE(!swapIndex_->forwardingTermStructure().empty(),
                        "null term structure set to this instance of " <<
                        swapIndex_->name());
-            ext::shared_ptr<VanillaSwap> temp =
-                swapIndex_->underlyingSwap(fixingDate_);
+            ext::shared_ptr<FixedVsFloatingSwap> temp = swapIndex_->underlyingSwap(fixingDate_);
```

Comment:
`underlyingSwap` is not virtual — I think we should check if we need `swapIndex_->underlyingSwap` or `OIswap_index->underlyingSwap`